### PR TITLE
feat: Edit masks

### PR DESCRIPTION
**Pull request by timothee-LJN**
## Description

Fixes pixano#24

### CODE REVIEW

**Review by cpvannier — CHANGES_REQUESTED** [review]
- [ ] When saving a polygon as a mask or editing a mask, the conversion from SVG to RLE breaks
- [x] When clicking the edit icon again to validate changes on a mask, cannot save as "Save changes" is still greyed out (this also happens when trying to edit bounding boxes) : pixano/pixano-project-manager#13
- [x] When creating a polygon on a dataset without objects, object is saved but nothing is displayed on the image
- [x] Cannot create a concave polygon with current method : pixano/pixano-project-manager#15

**Comment by timothee-LJN** [comment]
I've created [this issue](https://github.com/pixano/pixano-project-manager/issues/13) to improve "save changes" toggling issue. 
We are still working on fixing mask glitch !

File: ui/components/canvas2d/src/components/PolygonGroup.svelte [hunk 1]
```diff
@@ -87,43 +122,63 @@
...
+        <Circle
+          on:click={() => handlePolygonPointsClick(pi, viewId)}
+          on:dragmove={() => handlePolygonPointsDragMove(point.id, i)}
+          on:dragend={handlePolygonPointsDragEnd}
+          on:mouseover={() => point.id === 0 && updateCircleRadius(point.id, i, 4)}
+          on:mouseleave={() => updateCircleRadius(point.id, i, 2)}
+          config={{
+            x: point.x,
+            y: point.y,
+            radius: 2,
```
Comment (BertrandRenault) [inline ui/components/canvas2d/src/components/PolygonGroup.svelte:163]: Note for later: radius should be zoom dependant

File: ui/components/canvas2d/src/Canvas2D.svelte [hunk 2]
```diff
@@ -362,38 +362,38 @@
...
-            maskKonva.stroke(style.color);
-            maskKonva.fill(`rgba(${style.color.replace("rgb(", "").replace(")", "")}, 0.35)`);
-          }
-        }
-      }
-
-      destroyDeletedObjects(maskIds, maskGroup);
-    }
-  }
+  // function updateMasks(viewId: string) {
```
Comment (BertrandRenault) [inline ui/components/canvas2d/src/Canvas2D.svelte:365]: do we need to keep it ?

**Review by BertrandRenault — APPROVED** [review]

File: ui/components/canvas2d/src/components/PolygonGroup.svelte (hunk 1 above)
Comment (timothee-LJN) [reply ui/components/canvas2d/src/components/PolygonGroup.svelte:163]: yes, good idea. I add an issue.

File: ui/components/canvas2d/src/Canvas2D.svelte (hunk 2 above)
Comment (timothee-LJN) [reply ui/components/canvas2d/src/Canvas2D.svelte:365]: nope, i'll remove it and all dependencies !

File: ui/components/canvas2d/src/components/PolygonGroup.svelte (hunk 1 above)
Comment (timothee-LJN) [reply ui/components/canvas2d/src/components/PolygonGroup.svelte:163]: https://github.com/orgs/pixano/projects/3/views/1?pane=issue&itemId=50133596

**Review by cpvannier — APPROVED** [review]

**Pull request merged**